src/darwin/Framework/CHIP/MTRDeviceControllerDelegate.h

```diff
@@ -54,6 +54,18 @@ API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4))
  */
 - (void)controller:(MTRDeviceController *)controller commissioningComplete:(NSError * _Nullable)error;
 
+/**
+ * Notify the delegate when commissioning is completed.
+ * For getting commissioning controllee's deviceId
+ */
+- (void)controller:(MTRDeviceController *)controller
+    commissioningComplete:(NSError * _Nullable)error
+                 deviceId:(NSNumber * _Nullable)deviceId;
+
+/**
+ * Notify the delegate when read commissioning Infomation. (vendorID, productID)
```

Comment:
These (vendor id and product id) are already passed in to the device attestation delegate, right?

Reply:
Yes, it is. However, in order to use delegate to bring vendorID and productID, it is necessary to implement delegate. Also, it is called at a time earlier than when the delegate callback is called.

Reply:
Is there a case that could be spoofing? It seems that the information that simply allows you to know the VendorID and ProductID is all.

Reply:
Please don't resolve conversations that are not resolved.  Which this one clearly is not, since you asked me a question.

My point is that the vendorID/productID reported here might be misleading, in cases when the device would later fail attestation.  At a bare minimum, the limitations here need to be spelled out: it's very possible the device is _not_ in fact a device created by the vendor identified by vendorID, at this stage.

Reply:
Before adding an accurate comment, I have one question.
In what cases can the vendorID be different? I see that prior to this point I am getting information from the basicInformation Cluster.
(Please refer to the code below.)

https://github.com/project-chip/connectedhomeip/blob/7ccd5d40e4540e2a2ef1b24686a8128fe6c6d2ff/src/controller/CHIPDeviceController.cpp#L1890

https://github.com/project-chip/connectedhomeip/blob/7ccd5d40e4540e2a2ef1b24686a8128fe6c6d2ff/src/controller/CHIPDeviceController.cpp#L2047

Reply:
What can happen is that the device's Basic Information cluster lists a different vendor id from its actual attestation certificates.  If that happens, attestation will later fail, because the device is lying about its vendor id.  But at this stage no validation has been done on that.  So the value we have here is whatever came from the Basic Information cluster, but we have no confidence that it's actually a correct identifier for the entity that produced the device, at this point.

Reply:
I understand. I will update in document.

Reply:
I updated document. Please check about this.